src/api/java/com/minecolonies/api/entity/combat/combat/ThreatTable.java

```diff
@@ -0,0 +1,257 @@
+package com.minecolonies.api.entity.combat.combat;
+
+import net.minecraft.entity.LivingEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.minecolonies.api.util.constant.Constants.TICKS_SECOND;
+
+/**
+ * Threat table class, basically a list of entities with an associated threat value
+ */
+public class ThreatTable
+{
+    /**
+     * Melee range
+     */
+    private static final int MELEE_RANGE = 8;
+
+    /**
+     * Max tracking time
+     */
+    private static final int MAX_TRACKING_TICKS = TICKS_SECOND * 120;
+
+    /**
+     * List holding the threat entries
+     */
+    private List<ThreatTableEntry> threatList = new ArrayList<>();
+
+    /**
+     * Currently targeted entity
+     */
+    private int currentTargetIndex = 0;
+
+    /**
+     * The owner entity of this table
+     */
+    private final LivingEntity owner;
+
+    public ThreatTable(final LivingEntity owner)
+    {
+        this.owner = owner;
+    }
+
+    /**
+     * Adds X threat to the given entity
+     *
+     * @param attacker entity to add the value to
+     * @param threat   threat value to add
+     */
+    public void addThreat(final LivingEntity attacker, final int threat)
```

Comment:
maybe call it "threatModifier" instead of 'threat'

Reply:
additionalThreat then, modifier sounds like a multiplicator thing

Reply:
threatAddition, everything that is just "threat" seems like the entity (attacker) and not the addition to the threat level it gets.